rbc/remotejit.py

```diff
@@ -358,6 +359,15 @@ def get_signatures(self):
     def __call__(self, *arguments, **options):
         """Return the result of a remote JIT compiled function call.
         """
+        if not self.remotejit._supports_callable_caller:
+            msg = (
+                "Cannot call functions decorated by "
+                f"{self.remotejit.__class__}. These functions are "
+                "registered on the database and can be called "
+                "only from a SQL query."
```

Comment:
Nit: in general, there may exist other reasons for not supporting callable callers, and derivates of RemoteJIT may not be related to database applications. So, I suggest a more general exception message for the remotejit.py code:
```suggestion
                f"{type(self.remotejit)}."
```
and document this restriction in Omnisci related docs (to be created).

If we really want to add some explanation as above (which would be a nice feature for users not reading the docs) then the omniscidb.py could define, say, `OmnisciCaller` class derived from `Caller` with `__call__` method disabled and attach this to RemoteOmnisci class as `_caller_cls` attribute and let `RemoteJIT` use it whenever a `Caller` instance needs to be created. Let's come back to this when it is obvious that new users may hit this use case more often than expected.

Reply:
I did the change.

For possible future improvement with `OmnisciCaller`, I think a factory pattern would work good here. We could have factories for all these different classes.